agent/src/iconfucius/mcp_server.py

```
@@ -0,0 +1,146 @@
+                inputSchema=t["input_schema"],
+            )
+            for t in TOOLS
+        ]
+
+    @server.call_tool()
+    async def _call_tool(name: str, arguments: dict) -> CallToolResult:
+        try:
+            result = await asyncio.to_thread(
+                execute_tool, name, arguments, persona_name="iconfucius",
+            )
+        except BaseException as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `BaseException`

(BLE001)